Editor/Window/Components/StatusIndicator.cs

```diff
@@ -22,15 +23,33 @@ public StatusIndicator()
         {
             var uxml = Resources.Load<VisualTreeAsset>("EditorWindow/Components/StatusIndicator");
             uxml.CloneTree(this);
+            AddToClassList("status-indicator");
             AddToClassList("separator");
             AddToClassList("separator--horizontal");
         }
 
+        private void AddSpin()
+        {
+            schedule.Execute(() => { _imageContainer.AddToClassList("status-indicator--spin"); }).StartingIn(10);
+        }
+
         public void Set(State state, string text)
         {
             Reset();
             _label.text = text;
             AddToClassList(_stateClassNames[state]);
+            if (state == State.InProgress)
+            {
+                _imageContainer.RegisterCallback<TransitionEndEvent>(evt =>
```

Comment:
Doesn't this add multiple duplicate callbacks as the user continues to use the plugin window (anytime the update event is triggered when in progress this will be re-added)?

Is there a reason we aren't adding this callback once at construction?

Reply:
Fixed!